### PR TITLE
Add runable pipeline 

### PR DESCRIPTION
**Pull request by nhtlongcs**
This PR is to track the implementation progress

- [x] Document code 
- [x] Runnable code
- [x] Document README. Example for training, finetuning, prediction for segmentation and one for general task

### CODE REVIEW

File: nncore/models/loss/CEwithstat.py [hunk 1]
```diff
@@ -4,11 +4,11 @@
 import torch
 
 
-class CE(nn.Module):
-    """Some Information about CE"""
+class CEwithstat(nn.Module):
+    r"""CEwithstat is warper of cross-entropy loss"""
```
Comment (wan2000) [inline nncore/models/loss/CEwithstat.py:8]: I wonder if it is necessary to add `r` before the description (because I don't see any `r` in other descriptions).

**Review by wan2000 — APPROVED** [review]

File: nncore/models/loss/CEwithstat.py (hunk 1 above)
Comment (wan2000) [reply nncore/models/loss/CEwithstat.py:8]: Well, I think it does not relate to PR so I think it ok.
Just note this in the future for unifying.

Comment (nhtlongcs) [reply nncore/models/loss/CEwithstat.py:8]: I think finalize documentation in other PR is better idea, thanks for your opinion

**Review by VinhLoiIT — APPROVED** [review]
I think this PR is too big that we just accept for now since it is still in the early stage of development. Some refactor that should be made later includes (just note down for now):
- Unify raise NotImplementError/NotImplemented (they are used in different cases)
- Get rid of print and use logging instead
- Avoid `from X import *` for more clarity even in `__init__.py`
- README is too long to read, consider separate parts and move to the corresponding sub-directory and/or a document site
